source/common/event/dispatcher_impl.cc

```diff
@@ -26,13 +26,18 @@ namespace Envoy {
 namespace Event {
 
 DispatcherImpl::DispatcherImpl()
-    : DispatcherImpl(Buffer::WatermarkFactoryPtr{new Buffer::WatermarkBufferFactory}) {}
+    : DispatcherImpl(Buffer::WatermarkFactoryPtr{new Buffer::WatermarkBufferFactory}) {
+  // The dispatcher won't work as expected if libevent hasn't been configured to use threads.
+  RELEASE_ASSERT(Libevent::Global::initialized());
+}
 
 DispatcherImpl::DispatcherImpl(Buffer::WatermarkFactoryPtr&& factory)
     : buffer_factory_(std::move(factory)), base_(event_base_new()),
       deferred_delete_timer_(createTimer([this]() -> void { clearDeferredDeleteList(); })),
       post_timer_(createTimer([this]() -> void { runPostCallbacks(); })),
-      current_to_delete_(&to_delete_1_) {}
+      current_to_delete_(&to_delete_1_) {
+  ASSERT(Libevent::Global::initialized());
```

Comment:
Oh, this one should probably also be RELEASE_ASSERT

Reply:
What are you talking about, I already changed that file! ... *facepalm*  :-P